test/upload-test-images.sh

```diff
@@ -19,28 +19,37 @@ set -o errexit
 # If gcloud is not available make it a no-op, not an error.
 which gcloud &> /dev/null || gcloud() { echo "[ignore-gcloud $*]" 1>&2; }
 
+function upload_single_test_image() {
+  local docker_tag=$1
+  local image_name=$2
+  local image="knative.dev/eventing/test/test_images/${image_name}"
+  local registry=${KO_DOCKER_REPO%/*}
+
+  ko publish -B ${image}
+
+  if [ -n "$docker_tag" ]; then
+      if [ "$registry" = "docker.io" ]; then
+          ko publish -B -t ${docker_tag} ${image}
+      else
+          image=$KO_DOCKER_REPO/${image_name}
```

Comment:
What's the use case (registry) for gcloud? If it's for gcr.io, then let's get rid of this.

Reply:
Done.